### ISSUE TITLE
Cannot use redirect option with negated domain

### ISSUE THREAD

**Issue by Jtasiu** [uBlockOrigin/uBlock-issues#310]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

I am writing some filters but I noticed the below kind didn't work

`*pop.js$important,script,redirect=noopjs,domain=~site.com`

Why negated domains don't work at all with redirect option? I would like to create filters to do this as I don't mind performance hit, can deal with breakage myself and would rather redirect some file names totally. Is it possible to add a way to do this?

**Comment by gwarser** [comment]
Seems to work for `a.com|~b.com` and other way around, but not for simple `~a.com`

Tested on `https://www.wykop.pl/` and `https://www.wp.pl/` with variants of `*/xgemius.js$script,redirect=noopjs,domain=wp.pl|~wykop.pl`

**Comment by gwarser** [comment]
See also https://github.com/gorhill/uBlock/issues/3590

**Comment by gorhill** [comment]
The issue is that `redirect=` should have been made to strictly create redirection directives, not blocking filters -- I keep repeating this. The issue here can be solved only if changing all the semantic of the `redirect=` filters, which of course can't happen. OP's issue can be resolved by overriding the blanket redirection with another filter.

**Comment by uBlock-user** [comment]
> The issue here can be solved only if changing all the semantic of the redirect= filters, which of course can't happen.

So this is a won't fix ?

**Comment by gorhill** [comment]
> So this is a won't fix ?

I could made the filter `*pop.js$important,script,redirect=noopjs,domain=~site.com` work, but it would have to be understood that if ever a network request of type `script` from `site.com` pattern-matching `pop.js` is ever blocked by whatever mean, it would be redirected to `noopjs`.

**Comment by uBlock-user** [comment]
You mean `except` from site.com as thats negated.

**Comment by gorhill** [comment]
I meant what I wrote.

**Comment by uBlock-user** [comment]
Why would it apply on the negated domain ?

**Comment by gwarser** [comment]
Filter will be split into two parts: blocking part and redirecting. Negation will disable blocking. Blocking by any other means/filters will trigger redirection.

**Comment by uBlock-user** [comment]
Wouldn't negation affect redirection too ? If that's the case, then there's no point in adding a negated domain.

**Comment by Jtasiu** [comment]
@gwarser 
>Seems to work for a.com|~b.com and other way around, but not for simple ~a.com (no redirection at all)

Don't think the filter you wrote works for negated...to me looks like a.com will follow the filter rule but the negated domain will not work with the filter (it will be discarded). 

@gorhill
>OP's issue can be resolved by overriding the blanket redirection with another filter.

Not sure what that another filter is...please post example?

**Comment by prateekrastogi** [comment]
Despite multiple issues to fix Bing Ads, there has been no response whatsoever and agitation from repo maintainers:
So, referencing my issue here:
https://github.com/uBlockOrigin/uAssets/issues/6887#issuecomment-578509029

Let's name and shame ublock origin until their ego issues get resolved.

**Comment by gorhill** [comment]
I am going to decline this, using negated domains with redirect option is ambiguous. In case of ambiguity, use two separate filters, one for blocking, one for redirecting with `redirect-rule=`.

**Comment by gwarser** [comment]
How to create global redirect rule and then disable it on one domain?

**Comment by gwarser** [comment]
```
assets/wiki-b38a80ef.js$script,redirect=noop.js
assets/wiki-b38a80ef.js$script,redirect=none,domain=github.com
```

**Comment by uBlock-user** [comment]
```
assets/wiki-b38a80ef.js$script,redirect=noop.js
assets/wiki-b38a80ef.js$script,redirect-rule=none,domain=github.com
```

Like that

**Comment by gwarser** [comment]
Should be fixed by https://github.com/gorhill/uBlock/commit/157cef6034a8ec926c1e59c7e77f0a1fcbef473c
